hwcomposer/qeglfsscreen.cpp

```diff
@@ -99,4 +128,43 @@ qreal QEglFSScreen::refreshRate() const
     return m_hwc->refreshRate();
 }
 
+#ifdef WITH_SENSORS
+void QEglFSScreen::orientationReadingChanged()
+{
+    QSize screenSize = m_hwc->screenSize();
+    Qt::ScreenOrientation screenPrimaryOrientation = Qt::PortraitOrientation;
+    if (screenSize.width() > screenSize.height()) {
+        screenPrimaryOrientation = Qt::LandscapeOrientation;
+    }
+
+    QOrientationReading *orientationReading = m_orientationSensor->reading();
+    QOrientationReading::Orientation currentOrientation = orientationReading->orientation();
+
+    switch (currentOrientation) {
+    case QOrientationReading::TopUp:   /* 0 */
+        m_screenOrientation = screenPrimaryOrientation;
+        break;
+    case QOrientationReading::LeftUp:  /* 90 clockwise */
+        m_screenOrientation = screenPrimaryOrientation == Qt::PortraitOrientation ? 
+        Qt::InvertedLandscapeOrientation : Qt::PortraitOrientation;
```

Comment:
Sorry to nitpick about formatting but maybe add some indentation (add the usual 4 spaces) to the second line in these so it's clear these are part of the previous line.

Reply:
Fixed